compass/landice/tests/ismip6_forcing/ocean_thermal/process_thermal_forcing.py

```diff
@@ -0,0 +1,347 @@
+        to the ones that MALI uses.
+
+        Parameters
+        ----------
+        remapped_file_temp : str
+            temporary ismip6 data remapped on mali mesh
+
+        output_file : str
+            remapped ismip6 data renamed on mali mesh
+        """
+
+        config = self.config
+        process_obs_data = self.process_obs
+
+        # open dataset in 20 years chunk
+        ds = xr.open_dataset(remapped_file_temp, chunks=dict(time=20),
+                             engine="netcdf4")
+
+        ds["ismip6shelfMelt_zOcean"] = ds.z
+        ds = ds.drop_vars("z")  # dropping 'z' while it's still called 'z'
+
+        # build dictionary for ismip6 variables that MALI takes in
+        if process_obs_data:
+            ismip6_to_mali_dims = dict(
+                z="nISMIP6OceanLayers",
+                ncol="nCells")
+            ds["xtime"] = ("Time", ["2015-01-01_00:00:00".ljust(64)])
+            ds["xtime"] = ds.xtime.astype("S")
+            ds["thermal_forcing"] = ds["thermal_forcing"].expand_dims(
+                dim="Time", axis=0)
+            ds = ds.rename(ismip6_to_mali_dims)
+        else:
+            ismip6_to_mali_dims = dict(
+                z="nISMIP6OceanLayers",
+                time="Time",
+                ncol="nCells")
+            ds = ds.rename(ismip6_to_mali_dims)
+            # add xtime variable
+            xtime = []
+            for t_index in range(ds.sizes["Time"]):
+                date = ds.Time[t_index]
+                # forcing files do not all match even years,
+                # so round up the years
+                # pandas round function does not work for years,
+                # so do it manually
+                yr = date.dt.year.values
+                mo = date.dt.month.values
+                dy = date.dt.day.values
+                dec_yr = np.around(yr + (30 * (mo - 1) + dy) / 365.0)
+                date = f"{dec_yr.astype(int)}-01-01_00:00:00".ljust(64)
+                xtime.append(date)
+
+            ds["xtime"] = ("Time", xtime)
+            ds["xtime"] = ds.xtime.astype('S')
+            ds = ds.drop_vars(["Time"])
+
+        ismip6_to_mali_vars = dict(
+            thermal_forcing="ismip6shelfMelt_3dThermalForcing")
+        ds = ds.rename(ismip6_to_mali_vars)
+
+        # drop unnecessary variables
+        ds = ds.drop_vars(["z_bnds", "lat_vertices", "area",
+                           "lon_vertices", "lat", "lon"])
+
+        # transpose dimension
+        ds["ismip6shelfMelt_3dThermalForcing"] = \
+            ds["ismip6shelfMelt_3dThermalForcing"].transpose(
+            "Time", "nCells", "nISMIP6OceanLayers")
+
+        # write to a new netCDF file
+        write_netcdf(ds, output_file)
+        ds.close()
+
+    # create a nested dictionary for the ISMIP6 original forcing files including relative path
+    _file_obs = ["AIS/Ocean_Forcing/climatology_from_obs_1995-2017/obs_thermal_forcing_1995-2017_8km_x_60m.nc"]
```

Comment:
@trhille, this is a good point, and it would be ideal to have a case-insensitive directory for the ocean forcing files. The observational climatology file was originally only downloadable from the ISMIP6-2100 GHub endpoint, which is why I put the obs file in the same directory name `Ocean_Forcing` as that for the 2100 files. But the ISMIP6 committee has recently uploaded the obs data to the 2300 Ghub endpoint as well. I can see how to make the directory for the obs file case insensitive depending on which `period_endyear` is chosen in the config file in the following PRs I will create after merging this one.